scripts/licenses/main.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/build"
+	"os"
+	"strings"
+
+	"github.com/google/trillian/scripts/licenses/licenses"
+
+	"bitbucket.org/creachadair/shell"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var (
+	// Flags shared between subcommands
+	confidenceThreshold float64
+	buildTags           string
+
+	rootCmd = &cobra.Command{
+		Use: "licenses",
+	}
+)
+
+func init() {
+	rootCmd.PersistentFlags().Float64Var(&confidenceThreshold, "confidence_threshold", 0.9, "Minimum confidence required in order to positively identify a license.")
+	// Go build flags, which should match flags offered by `go build`
+	rootCmd.PersistentFlags().StringVar(&buildTags, "tags", "", "A space-separated list of build tags to consider satisfied.")
+}
+
+func main() {
+	rootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	if err := parseGoBuildFlags(rootCmd.PersistentFlags()); err != nil {
+		glog.Error(err)
+	}
+
+	if err := rootCmd.Execute(); err != nil {
+		glog.Exit(err)
+	}
+}
+
+// Libraries returns the libraries used by the package identified by importPath.
+// The import path is assumed to be in the context of the current working
+// directory, so vendoring and relative import paths will work.
+func libraries(importPath string) ([]*licenses.Library, error) {
+	// Import the main package and find all of the libraries that it uses.
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	pkg, err := build.Import(importPath, wd, build.ImportMode(0))
+	if err != nil {
+		return nil, err
+	}
+	buildCtx := build.Default
+	buildCtx.BuildTags = append(buildCtx.BuildTags, strings.Split(buildTags, " ")...)
```

Comment:
Is it OK to modify `build.Default`?  Presumably there's nothing else using it concurrently.

Reply:
Just in case, I've changed this to overwrite BuildTags with the output of `strings.Split()` - there shouldn't be any BuildTags set on `build.Default` anyway, and it's not clear we'd want to apply them even if there were, given that we've got a dedicated flag for setting build tags.

Reply:
Just in case, I've changed this to overwrite BuildTags with the output of strings.Split() - there shouldn't be any BuildTags set on build.Default anyway, and it's not clear we'd want to apply them even if there were, given that we've got a dedicated flag for setting build tags.